tests/blast/test_blast_output.h

```diff
@@ -0,0 +1,814 @@
+                          "Query_Numero_Uno\tSubject_Numero_Dos\t100.00\t8\t0\t0\t3\t10\t157\t164\t0.001\t22.3\n");
+    }
+    else if (customFields == 1)
+        compString.append(
+        "Query_Numero_Uno\tSubject_Numero_Uno\t71.43\t14\t2\t1\t1\t12\t79\t92\t5e-04\t23.1\t48\t0/0\n"
+        "Query_Numero_Uno\tSubject_Numero_Dos\t100.00\t8\t0\t0\t3\t10\t157\t164\t0.001\t22.3\t46\t0/0\n");
+    else
+        compString.append(
+        "Query_Numero_Uno with args\tSubject_Numero_Uno\t14\t2\t2\t0.000534696\t23.0978\n"
+        "Query_Numero_Uno with args\tSubject_Numero_Dos\t8\t0\t0\t0.000912053\t22.3274\n");
+
+    if ((getBlastTabularSpec(context) == BlastTabularSpec::HEADER) && (format > 0))
+    {
+        // header for empty record
+        compString.append(versionLine);
+        compString.append("# Query: Query_Numero_Dos with args\n"
+                          "# Database: The Foo Database\n");
+        if (context.legacyFormat)
+            compString.append(fieldsLine);
+        if (!context.legacyFormat)
+            compString.append("# 0 hits found\n");
+
+        // header for
+        compString.append(versionLine);
+        compString.append("# Query: Query_Numero_Tres with args\n"
+                          "# Database: The Foo Database\n");
+        compString.append(fieldsLine);
+        if (!context.legacyFormat)
+            compString.append("# 2 hits found\n");
+    }
+
+    if (customFields == 0)
+        compString.append("Query_Numero_Tres\tSubject_Numero_Dos\t100.00\t8\t0\t0\t1\t8\t10\t17\t0.007\t18.9\n"
+                          "Query_Numero_Tres\tSubject_Numero_Uno\t87.50\t8\t1\t0\t1\t8\t184\t191\t0.026\t16.9\n");
+    else if (customFields == 1)
+        compString.append("Query_Numero_Tres\tSubject_Numero_Dos\t100.00\t8\t0\t0\t1\t8\t10\t17\t0.007\t18.9\t37\t0/0\n"
+                          "Query_Numero_Tres\tSubject_Numero_Uno\t87.50\t8\t1\t0\t1\t8\t184\t191\t0.026\t16.9\t32\t0/0\n");
+    else
+        compString.append("Query_Numero_Tres with args\tSubject_Numero_Dos\t8\t0\t0\t0.00672262\t18.8606\n"
+                          "Query_Numero_Tres with args\tSubject_Numero_Uno\t8\t1\t0\t0.0255459\t16.9346\n");
+
+    if ((getBlastTabularSpec(context) == BlastTabularSpec::HEADER) && (!context.legacyFormat) && (format >= 1))
+        compString.append("# BLAST processed 3 queries\n");
+
+    if (contents != compString)
+    {
+        for (uint32_t i = 0; i < length(contents); ++i)
+        {
+            if (contents[i] != compString[i])
+            {
+                std::cout << contents.substr(0,i) << "\n";
+                std::cout << "CONT: \"" << contents[i] << "\"\n";
+                std::cout << "COMP: \"" << compString[i] << "\"\n";
+                break;
+            }
+        }
+    }
+    SEQAN_ASSERT_EQ(contents, compString);
+}
+
+// TEST MATCH, WITHOUT HEADER, DEFAULTS FIELDS
+
+SEQAN_DEFINE_TEST(test_blast_write_match_tabular_run_time_context_args)
```

Comment:
I haven't done testing on this scale before, can you point me to the relevant documents describing how to do this in a nicer way?